src/functionalTest/java/uk/gov/hmcts/reform/bulkscanprocessor/controllers/TestHelper.java

```diff
@@ -140,24 +148,43 @@ public String getRandomFilename(String suffix) {
     ) throws IOException {
         ByteArrayOutputStream outputStream = new ByteArrayOutputStream();
         try (ZipOutputStream zos = new ZipOutputStream(outputStream)) {
+            Map<String, String> tokens = new HashMap<>();
+
             for (String file : files) {
-                zos.putNextEntry(new ZipEntry(file));
+                String dcn = generateDcnNumber();
+                tokens.put("$$" + file.replace(".pdf", "") + "$$", dcn);
+                zos.putNextEntry(new ZipEntry(dcn + ".pdf"));
                 zos.write(Resources.toByteArray(Resources.getResource(file)));
                 zos.closeEntry();
             }
 
             if (metadataFile != null) {
                 String metadataTemplate =
                     Resources.toString(Resources.getResource(metadataFile), StandardCharsets.UTF_8);
-                String metadata = metadataTemplate.replace("$$zip_file_name$$", zipFilename);
                 zos.putNextEntry(new ZipEntry("metadata.json"));
-                zos.write(metadata.getBytes());
+                tokens.put("$$zip_file_name$$", zipFilename);
+
+                Pattern pattern = Pattern.compile("(" + String.join("|", tokens.keySet()) + ")");
+                Matcher matcher = pattern.matcher(metadataTemplate);
+                StringBuffer sb = new StringBuffer();
+
+                while (matcher.find()) {
+                    matcher.appendReplacement(sb, tokens.get(matcher.group(1)));
+                }
+
+                matcher.appendTail(sb);
+
+                zos.write(sb.toString().getBytes());
                 zos.closeEntry();
             }
         }
         return outputStream.toByteArray();
     }
 
+    private String generateDcnNumber() {
+        return DateTimeFormatter.ofPattern("yyyyMMDDHHmmss").format(Instant.now()) + RANDOM.nextInt();
```

Comment:
`documentControlNumber VARCHAR(100)` so maybe just UUID?

Reply:
no real preference.. can do uuid, can do time based. uuid is more of a "real life identifier" while this one is clearly test purpose

Reply:
only digits allowed in json schema, nevermind